dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -121,8 +121,12 @@ class GaussianBlurOpCpu : public OpImplBase<CPUBackend> {
   using Kernel = kernels::SeparableConvolutionCpu<Out, In, float, axes, has_channels>;
   static constexpr int ndim = Kernel::ndim;
 
-  explicit GaussianBlurOpCpu(const OpSpec& spec, const DimDesc& dim_desc)
-      : spec_(spec), dim_desc_(dim_desc) {}
+  /**
+   * @param spec  Pointer to a persistent OpSpec object,
+   *              which is guaranteed to be alive for the entire lifetime of this object
+   */
+  explicit GaussianBlurOpCpu(const OpSpec* spec, const DimDesc& dim_desc)
```

Comment:
Why change it to pointer just to take address at the call site and dereference it here?

Reply:
See the comment :)

Reply:
That's basically a definition of what pass by reference means.

Reply:
Imagine:

```
GaussianBlurOpCpu impl_;
ParentOp(const OpSpec &spec, ...) 
  : impl_ (spec) {}

// in a function
OpSpec spec;
...
return ParentOp(spec);
```
^^ Looks legit from here, but it's actually storing a reference to a stack variable

```
GaussianBlurOpCpu impl_;
ParentOp(const OpSpec &spec, ...) 
  : impl_ (&spec) {}
 ```
^^ If I have to write here, it becomes obvious that I am doing something wrong, taking the address of a ctor argument.



```
GaussianBlurOpCpu impl_;
ParentOp(const OpSpec &spec, ...) 
  : spec_(spec), impl_ (&spec_) {}
 ```
 ^^ And would make it more obvious that I should pass a pointer to a persistent object

Reply:
In other words, I changed the internal impl to take a pointer to make it obvious that it's storing a view of the OpSpec only